sql/core/src/test/scala/org/apache/spark/sql/sources/DataSourceTest.scala

```diff
@@ -20,14 +20,17 @@ package org.apache.spark.sql.sources
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql._
 import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.UTF8String
 
 private[sql] abstract class DataSourceTest extends QueryTest {
 
-  protected def sqlTest(sqlString: String, expectedAnswer: Seq[Row]) {
+  protected def sqlTest(sqlString: String, expectedAnswer: Seq[Row], enableRegex: String = "true") {
```

Comment:
`enableRegex: String = "true"` -> `enableRegex: Boolean = false`

Could you change the type to Boolean and call .toString below and set the default to `false`?